mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -2231,10 +2218,14 @@ static func::FuncOp createVerifierFunc(ModuleOp module, const KernelIF &kernel,
                            ValueRange{testResult, valResult, printDebugVal});
   }
 
-  if (!isTestAndValSameType) {
+  if (testResultNew && !testResultNew.use_empty()) {
```

Comment:
If we allocated a new buffer, and it's still in use, deallocate it.  Same effect as checking the original flag.